src/hotspot/share/opto/superword.cpp

```diff
@@ -1742,7 +1744,21 @@ VTransformBoolTest PackSet::get_bool_test(const Node_List* bool_pack) const {
       mask = bol->_test.negate();
       is_negated = true;
     }
-  }
+    break;
+  }
+  case Op_CmpU:
+  case Op_CmpUL:
+    // Carry unsigned-ness information from CmpUxx to VTransformBoolTest,
+    // which will be passed to e.g. VectorMaskCmp.
+    mask = BoolTest::unsigned_mask(mask);
+    break;
+  case Op_CmpI:
+  case Op_CmpL:
+    break;
```

Comment:
We could add a comment why these are ok without any action.
```suggestion
    // The mask of signed int/long scalar comparisons has the same semantics
    // as the mask for vector elementwise int/long comparison with VectorMaskCmp.
    break;
```

Reply:
Nice! I can commit your suggestion by simply click the button.